### ISSUE TITLE
React Router how to use charting-libuary?

### ISSUE THREAD

**Issue by kenneth** [tradingview/charting-library-examples#136]
React Router will redirect static file  `charting_library`  folder and 404 error
Any suggestions?

**Comment by timocov** [comment]
Maybe you can provide some steps to reproduce? Without them I don't think and somebody can help you with that.

**Comment by kenneth** [comment]
@timocov just like this ,react route can't deal with static file and make charting_library route can't access ,404 error

http://binance.quomap.com/trade/BTCUSDT
https://binance.quomap.com/trade/BTCUSDT

`
import React from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'

const App = () => (
  <Router>
    <React.Fragment>
                    <Route exact path="/" component={MarketPairs} />
                    <Route path="/trade/:symbol" component={Trade} />
    </React.Fragment>
  </Router>
);  

export default App;
`

**Comment by timocov** [comment]
When I said "step to reproduce" I mean step-by-step guide to run the issue locally (repo with the example and steps how to run/get the issue).

> react route can't deal with static file and make charting_library 

Can you serve library's files not by react router but with any other server (like nginx and so on)? All you need to bundle `charting_library.min` file inside your js bundle (it's API) and then serve library's files somehow (you don't need to serve them with react-router because it's not part of your bundle and you don't need to process them anyhow). But maybe I don't understand you correctly.

**Comment by linh626688** [comment]
Hi @timocov I think I have issues with react-router and charting_library too. 
I started with CRA, I work fine, but when adding react-router and Page have ChartComponent located in a route ( /chart ), the Chart didn't render when I changed route (/chart to /home, to /chart). I inspected src on browser and not see <iframe> elenmet

**Comment by timocov** [comment]
How react-router works? Does it detach a node from DOM or remove it completely?

**Comment by linh626688** [comment]
> How react-router works? Does it detach a node from DOM or remove it completely?

As I know, React-router update a node from DOM

**Comment by timocov** [comment]
@linh626688 Thus it removes iframe from the tree? It might be the issue, I think you need to patch router somehow to avoid removing iframe (or any its parent node) from the tree because detaching (replacing/removing) iframe from DOM automatically unload it.

**Comment by linh626688** [comment]
oh, I'm not sure about that. Do you have a sample or suggestion?

**Comment by timocov** [comment]
@linh626688 No, unfortunately I'm not familiar with react-related stack much

**Comment by Hawkeyeanalytics** [comment]
Add  '/charting_library' to your allowed urls
